packages/ember-glimmer/lib/utils/lookup-component.js

```diff
@@ -7,14 +8,40 @@ function lookupComponentPair(componentLookup, owner, name, options) {
   let component = componentLookup.componentFor(name, owner, options);
   let layout = componentLookup.layoutFor(name, owner, options);
 
-  if (!layout && component) {
-    let layoutProp = component.proto().layout;
-    if (layoutProp) {
-      let templateFullName = 'template:components/' + name;
-      owner.register(templateFullName, layoutProp);
+  if (component) {
+    let templateFullName = 'template:components/' + name;
+    let {
+      templateForName,
+      layout: layoutProp,
+      layoutName: layoutNameProp,
+      defaultLayout: defaultLayoutProp
+    } = component.proto();
+
+    if (defaultLayoutProp && !layoutProp) {
+      // If a `defaultLayout` was specified move it to the `layout` prop.
+      // `layout` is no longer a CP, so this just ensures that the `defaultLayout`
+      // logic is supported with a deprecation
+      deprecate(
+        `Specifying \`defaultLayout\` to ${component} is deprecated. Please use \`layout\` instead.`,
+        false,
+        {
+          id: 'ember-views.component.defaultLayout',
+          until: '3.0.0',
+          url: 'http://emberjs.com/deprecations/v2.x/#toc_ember-component-defaultlayout'
+        }
+      );
+
+      owner.register(templateFullName, defaultLayoutProp);
       layout = owner.lookup(templateFullName, options);
-    } else {
-      layout = owner.lookup(DEFAULT_LAYOUT);
+    } else if (!layout) {
+      if (layoutProp) {
+        owner.register(templateFullName, layoutProp);
+        layout = owner.lookup(templateFullName, options);
+      } else if (layoutNameProp) {
+        layout = templateForName(layoutNameProp, null, owner);
```

Comment:
this is weird, lets just move the templateForName logic to the mixin that is actually calling it.